src/Services/LinkTypeService.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace SilverStripe\LinkField\Services;
+
+use InvalidArgumentException;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\LinkField\Models\Link;
+
+class LinkTypeService
+{
+    use Injectable;
+
+    /**
+     * Generate all link types that are subclasses of Link::class
+     */
+    public static function generateAllLinkTypes(): array
```

Comment:
Generally, when we make an injectable service, we make it's method instance method, rather than static.

```suggestion
    public function generateAllLinkTypes(): array
```

Reply:
Updated